cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -148,7 +149,42 @@ private ConfigResponse createResponse(Application app)
         }
       }
 
-      app.configure(configurer, new DefaultApplicationContext<>(appConfig));
+      try {
+        app.configure(configurer, new DefaultApplicationContext<>(appConfig));
+      } catch (Throwable t) {
+        Throwable rootCause = Throwables.getRootCause(t);
+        if (rootCause instanceof ClassNotFoundException) {
+          // Heuristic to provide better error message
+          String missingClass = rootCause.getMessage();
+
+          // If the missing class has "spark" in the name, try to see if Spark is available
+          if (missingClass.startsWith("org.apache.spark.") || missingClass.startsWith("co.cask.cdap.api.spark.")) {
+            File sparkAssemblyJar;
+            try {
+              sparkAssemblyJar = SparkUtils.locateSparkAssemblyJar();
+            } catch (Exception e) {
+              // Spark is not available, it is most likely caused by missing Spark in the platform
+              throw new IllegalStateException(
+                "Missing Spark related class " + missingClass +
+                  ". It may be caused by unavailability of Spark in the platform.", t);
```

Comment:
"platform" is a little hard to understand for users. Is what you mean that Spark is not installed in the cluster? Or that SPARK_HOME is not set? Perhaps it is not set in cdap-env.sh?